tests/cross/src/test/scala/tests/pc/SignatureHelpPatternSuite.scala

```diff
@@ -91,8 +85,113 @@ class SignatureHelpPatternSuite extends BaseSignatureHelpSuite {
       |  }
       |}
       |""".stripMargin,
-    """|unapply(a: Int): Option[(Int, Int)]
-       |        ^^^^^^
+    """|(Int, Int)
+       | ^^^
        |""".stripMargin
   )
+
+  check(
+    "non-synthetic-unapply-second",
+    """
+      |class HKT[C[_], T](a: C[T])
+      |object HKT {
+      |  def unapply(a: Int): Option[(Int, Int)] = Some(2 -> 2)
+      |}
+      |object Main {
+      |  (null: Any) match {
+      |    case HKT(1, @@) =>
+      |  }
+      |}
+      |""".stripMargin,
+    """|(Int, Int)
+       |      ^^^
+       |""".stripMargin
+  )
+
+  check(
+    "pat",
+    """
+      |case class Person(name: String, age: Int)
+      |object a {
+      |  null.asInstanceOf[Person] match {
+      |    case Person(@@)
+      |}
+    """.stripMargin,
+    """|(name: String, age: Int)
+       | ^^^^^^^^^^^^
+       | """.stripMargin
+  )
+
+  check(
+    "pat1",
+    """
+      |class Person(name: String, age: Int)
+      |object Person {
+      |  def unapply(p: Person): Option[(String, Int)] = ???
+      |}
+      |object a {
+      |  null.asInstanceOf[Person] match {
+      |    case Person(@@) =>
+      |  }
+      |}
+    """.stripMargin,
+    """|(name: String, age: Int)
+       | ^^^^^^^^^^^^
+       | """.stripMargin
+  )
+
+  check(
+    "pat2",
+    """
+      |object a {
+      |  val Number = "$a, $b".r
+      |  "" match {
+      |    case Number(@@)
+      |  }
+      |}
+    """.stripMargin,
+    """|(String)
+       | ^^^^^^
+       |(Char)
+       |""".stripMargin
+  )
+
+  check(
+    "pat3",
+    """
+      |object And {
+      |  def unapply[A](a: A): Some[(A, A)] = Some((a, a))
+      |}
+      |object a {
+      |  "" match {
+      |    case And("", s@@)
+      |  }
+      |}
+  """.stripMargin,
+    """|(A, A)
+       |    ^
+       | """.stripMargin
+  )
+
+  check(
+    "pat4",
+    """
+      |object & {
+      |  def unapply[A](a: A): Some[(A, A)] = Some((a, a))
+      |}
+      |object a {
+      |  "" match {
+      |    case "" & s@@
+      |  }
+      |}
```

Comment:
This looks strange

Reply:
The case was added by Olaf a while ago, I haven't yet fixed it